datafusion/physical-expr/src/aggregate/covariance.rs

```diff
@@ -253,33 +253,33 @@ impl Accumulator for CovarianceAccumulator {
         let mut arr1 = downcast_value!(values1, Float64Array).iter().flatten();
         let mut arr2 = downcast_value!(values2, Float64Array).iter().flatten();
 
-        for _i in 0..values1.len() {
-            let value1 = arr1.next();
-            let value2 = arr2.next();
+        for i in 0..values1.len() {
+            let value1 = match values1.is_valid(i) {
+                true => arr1.next(),
+                false => None,
+            };
+            let value2 = match values2.is_valid(i) {
+                true => arr2.next(),
+                false => None,
+            };
 
             if value1.is_none() || value2.is_none() {
-                if value1.is_none() && value2.is_none() {
-                    continue;
-                } else {
-                    return Err(DataFusionError::Internal(
-                        "The two columns are not aligned".to_string(),
-                    ));
-                }
-            } else {
-                let value1 = unwrap_or_internal_err!(value1);
-                let value2 = unwrap_or_internal_err!(value2);
-                let new_count = self.count + 1;
-                let delta1 = value1 - self.mean1;
-                let new_mean1 = delta1 / new_count as f64 + self.mean1;
-                let delta2 = value2 - self.mean2;
-                let new_mean2 = delta2 / new_count as f64 + self.mean2;
-                let new_c = delta1 * (value2 - new_mean2) + self.algo_const;
-
-                self.count += 1;
-                self.mean1 = new_mean1;
-                self.mean2 = new_mean2;
-                self.algo_const = new_c;
+                continue;
             }
+
+            let value1 = unwrap_or_internal_err!(value1);
+            let value2 = unwrap_or_internal_err!(value2);
```

Comment:
Had we migrated to Rust 1.65, I would suggest the much simpler:
```
let Some(value1) = value1 else { continue };
let Some(value2) = value2 else { continue };
```
@alamb, any information as to when we will make the switch?

Reply:
I am referring to the `is_none`-`continue` check and the `unwrap_or_internal_err!` usage. GH's line selection tool messes up the context -- sorry about that.

Reply:
Looks cleaner, and it seems we dont need this macro anymore

Reply:
But yeah, its quite new feature, and we should support 1.62, so I guess, I'll revert this change

Reply:
Though I still can get rid of unwraps in `retract_batch` using this macro 🤔

Reply:
Agreed, sounds good. We can switch to the new `let-else` construct when the project migrates to 1.65.